### PR TITLE
Add ability to pass storage_options to pandas.to_parquet. Update docs/CHANGELOG

### PR DESCRIPTION
**Pull request by iameskild**
Questions:
- Do we want to include docstrings for `_retry_args` (`pack_partitions_to_parquet`)?
    - **_Answer_**: Swap out `retrying` with `tenacity` in another PR.
- Do we want to pass `storage_options` to `pandas.to_parquet` call?
    - [Available as a parameter in pandas > 1.2](https://github.com/pandas-dev/pandas/blob/eccfe9097ecf5d0d5fd5437290e956757d27933e/pandas/core/frame.py#L2631)
    - See `write_partition` (`pack_partitions_to_parquet`)
    - **_Answer_**: Added the ability to pass storage_options to `pandas.to_parquet` (only for version >= 1.2)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,13 @@
+## Version 0.4.3
+
+Date: 2021-08-05
+
+This release primarily expands the optional arguments that can be passed to `to_parquet_dask`/`read_parquet_dask` ensuring that `storage_options` is successfully passed where needed. It also adds the ability to pass `storage_options` to `pandas.to_parquet` function (only for pandas > 1.2) and renames any reference to `fname` with `path` to align with the pandas's convention.
```
Comment (jbednar) [inline CHANGELOG.md:5]:
```suggestion
This release primarily expands the optional arguments that can be passed to `to_parquet_dask`/`read_parquet_dask` ensuring that `storage_options` is successfully passed where needed. It also adds the ability to pass `storage_options` to the `pandas.to_parquet` function (only for pandas > 1.2) and renames any reference to `fname` with `path` to align with the pandas convention.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,13 @@
...
+
+Bug fixes:
+- Update `validate_coerce_filesystem` to pass `storage_options` throught. ([#78](https://github.com/holoviz/spatialpandas/pull/78))
```
Comment (jbednar) [inline CHANGELOG.md:8]:
```suggestion
- Update `validate_coerce_filesystem` to pass `storage_options` through. ([#78](https://github.com/holoviz/spatialpandas/pull/78))
```

File: spatialpandas/io/__init__.py [hunk 3]
```diff
@@ -1,3 +1,3 @@
 from .parquet import (  # noqa
     read_parquet, read_parquet_dask, to_parquet, to_parquet_dask
-)
+)
```
Comment (jbednar) [inline spatialpandas/io/__init__.py:3]: Fix missing final newline:
```suggestion
)

```

**Review by jbednar — APPROVED** [review]
Looks good!

File: spatialpandas/io/parquet.py [hunk 4]
```diff
@@ -85,31 +99,54 @@ def _get_geometry_columns(pandas_metadata):
...
         **kwargs,
-    )
+    }
+
+    if PANDAS_GE_12:
+        to_parquet_args.update({"storage_options":storage_options})
+    else:
+        if filesystem is None:
+            filesystem = validate_coerce_filesystem(path, filesystem, storage_options)
+        to_parquet_args.update({"filesystem":filesystem})
```
Comment (brl0) [inline spatialpandas/io/parquet.py:128]:
```suggestion
        to_parquet_args.update({"storage_options": storage_options})
    else:
        if filesystem is None:
            filesystem = validate_coerce_filesystem(path, filesystem, storage_options)
            to_parquet_args.update({"filesystem": filesystem})
```

**Pull request merged**